packages/spectral/src/types/Inputs.ts

```diff
@@ -109,102 +109,110 @@ interface BaseInputField {
   required?: boolean;
 }
 
-export interface StringInputField extends BaseInputField {
-  /** Data type the InputField will collect. */
-  type: "string";
-  /** Collection type of the InputField */
-  collection?: InputFieldCollection;
-  /** Default value for this field. */
-  default?: unknown;
-  /** Dictates possible choices for the input. */
-  model?: InputFieldChoice[];
-  /** Clean function */
-  clean?: InputCleanFunction<this["default"]>;
-}
+type CollectionOptions<T> =
+  | SingleValue<T>
+  | ValueListCollection<T>
+  | KeyValueListCollection<T>;
 
-export interface DataInputField extends BaseInputField {
-  /** Data type the InputField will collect. */
-  type: "data";
+interface SingleValue<T> {
   /** Collection type of the InputField */
-  collection?: InputFieldCollection;
+  collection?: undefined;
   /** Default value for this field. */
-  default?: unknown;
-  /** Dictates possible choices for the input. */
-  model?: InputFieldChoice[];
-  /** Clean function */
-  clean?: InputCleanFunction<this["default"]>;
+  default?: T;
 }
 
-export interface TextInputField extends BaseInputField {
-  /** Data type the InputField will collect. */
-  type: "text";
+interface ValueListCollection<T> {
   /** Collection type of the InputField */
-  collection?: InputFieldCollection;
+  collection: "valuelist";
   /** Default value for this field. */
-  default?: unknown;
-  /** Dictates possible choices for the input. */
-  model?: InputFieldChoice[];
-  /** Clean function */
-  clean?: InputCleanFunction<this["default"]>;
+  default?: T[];
 }
 
-export interface PasswordInputField extends BaseInputField {
-  /** Data type the InputField will collect. */
-  type: "password";
+interface KeyValueListCollection<T> {
   /** Collection type of the InputField */
-  collection?: InputFieldCollection;
+  collection: "keyvaluelist";
   /** Default value for this field. */
-  default?: unknown;
-  /** Dictates possible choices for the input. */
-  model?: InputFieldChoice[];
-  /** Clean function */
-  clean?: InputCleanFunction<this["default"]>;
+  default?: { key: string; value: T }[];
 }
 
-export interface BooleanInputField extends BaseInputField {
-  /** Data type the InputField will collect. */
-  type: "boolean";
-  /** Collection type of the InputField */
-  collection?: InputFieldCollection;
-  /** Default value for this field. */
-  default?: unknown;
-  /** Dictates possible choices for the input. */
-  model?: InputFieldChoice[];
-  /** Clean function */
-  clean?: InputCleanFunction<this["default"]>;
-}
+export type StringInputField = BaseInputField &
+  CollectionOptions<string> & {
+    /** Data type the InputField will collect. */
+    type: "string";
+    /** Dictates possible choices for the input. */
+    model?: InputFieldChoice[];
+    /** Clean function */
+    clean?: InputCleanFunction<unknown>;
+  };
+
+export type DataInputField = BaseInputField &
+  CollectionOptions<Buffer> & {
```

Comment:
`data` is more often used for indicating larger data payloads or data passed by reference. Defaults are also delivered as strings without further massaging of the data so I don't think we can do anything other than `string` here as well, at least without platform changes.